src/utils/decorateWithScope.js

```diff
@@ -0,0 +1,22 @@
+const decorateWithScope = (scopes, actionCreators) => {
+  const asyncRegex = /async/;
+  return Object.entries(actionCreators).reduce((result, [actionName, actionCreator]) => {
+    if (asyncRegex.test(actionCreator.toString())) {
```

Comment:
Why the distinction between `async` and sync?

Reply:
Wow, it's been a while since I did this haha that's because the decorator should only be applied to async functions. This handles async requests for different environments.

This feature is a really specific use case and should only be used in a repo with multiple environments.